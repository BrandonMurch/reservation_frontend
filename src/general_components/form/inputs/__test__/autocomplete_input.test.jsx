// Dependencies
import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import AutoCompleteInput from '../autocomplete_input';
import userEvent from '@testing-library/user-event';

// Stylesheet
import style from '../../form.module.css';

describe('<AutoCompleteInput />', () => {
  const props = {
    style,
    possibleEntries: ['apple', 'banana', 'baboon'],
    name: 'InputBox',
    onBlur: jest.fn(),
    updateValue: jest.fn(),
    label: 'Autocomplete!',
  };

  beforeEach(() => {
    render(
      <AutoCompleteInput {...props} />,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const tree = create(
      <AutoCompleteInput {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display suggestions when the user types', () => {
    const input = screen.getByRole('textbox', { name: 'Autocomplete!' });
    userEvent.click(input);
    userEvent.type(input, 'ba');
    expect(screen.getByText('banana')).toBeInTheDocument();
    expect(screen.getByText('baboon')).toBeInTheDocument();
    expect(screen.queryByText('apple')).toBeNull();
  });
  it('should display suggestions when the user types', () => {
    const input = screen.getByRole('textbox', { name: 'Autocomplete!' });
    userEvent.click(input);
    userEvent.type(input, 'ba');
    const suggestion = screen.getByText('banana');
    userEvent.click(suggestion);
    expect(props.onBlur).toHaveBeenCalledWith({ value: 'banana' });
  });
});
