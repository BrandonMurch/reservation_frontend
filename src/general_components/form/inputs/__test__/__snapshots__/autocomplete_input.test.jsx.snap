// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoCompleteInput /> should match snapshot 1`] = `
<div
  className="inputGroup"
>
  <label
    className="labelText"
    htmlFor="InputBox"
  >
    Autocomplete!
  </label>
  <input
    className="input"
    id="InputBox"
    name="InputBox"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    pattern={null}
    placeholder="Autocomplete!"
    required={false}
    type="text"
    value=""
  />
</div>
`;
